lib/tasks/journeys_report.rake

```diff
@@ -2,60 +2,97 @@ require 'csv'
 require 'json'
 require 'aws-sdk-s3'
 
+def get_last_run(bucket_name)
+  s3 = Aws::S3::Resource.new
+  if s3.bucket(bucket_name).object('bookmark').exists?
+    bookmark = s3.bucket(bucket_name).object('bookmark').get
+    last_run = JSON.parse(bookmark.body.read)['last_run']
+    last_run_date = DateTime.parse(last_run)
+  else
+    last_run_date = DateTime.strptime('0', '%s')
+  end
+  last_run_date
+end
+
+def update_bookmark(bucket_name)
+  s3 = Aws::S3::Resource.new
+  bookmark = { 'last_run': DateTime.now.utc.to_s }
+  s3.bucket(bucket_name).put_object(key: 'bookmark', body: JSON[bookmark])
+end
+
 namespace :journeys do
   desc 'Exports a JSON report of all journeys'
   task json_report: :environment do
     timestamp = DateTime.now.utc.to_i
+    bucket_name = ENV['S3_REPORTING_BUCKET_NAME'] || raise('S3_REPORTING_BUCKET_NAME not defined')
+
+    if ENV['RAILS_ENV'].to_s == 'development' || ENV['RAILS_ENV'].to_s == ''
+      Aws.config.update(
+        endpoint: 'http://localhost:4572',
+        credentials: Aws::Credentials.new('fakeid', 'fakesecret'),
+        region: 'eu-west-2',
+        force_path_style: true,
+      )
+    end
```

Comment:
This is useful during local development and the `endpoint` cannot yet be overridden with an ENV variable. 
We can get rid of it later.

Reply:
👍